rj_gameplay/stp/utils/edict.py

```diff
@@ -0,0 +1,65 @@
+from typing import TypeVar, Generic, Dict, Type
+from collections.abc import MutableMapping
+
+ValueInterfaceT = TypeVar("ValueInterfaceT")
+
+ValueConcreteT = TypeVar("ValueConcreteT")
+
+
+class EKey(Generic[ValueConcreteT]):
+    __slots__ = ["concrete_cls"]
+
+    def __init__(self, cls: Type[ValueConcreteT]):
+        self.concrete_cls = cls
+
+    def value_t(self) -> Type[ValueConcreteT]:
+        return self.concrete_cls
+
+
+class EDict(Generic[ValueInterfaceT], MutableMapping):
```

Comment:
Honestly I have no idea what the `E` in the name here is intending to convey.

Reply:
Naming things is hard...maybe `TypedKeyDict`? Still a little bit weird but I think it gets the point across a little better.

Reply:
Sure.